### PR TITLE
TrecQrels class

### PR DESCRIPTION
**Pull request by x65han**
- Introduce TrecQrels class so we can retrieve 1s 2s quickly.

### CODE REVIEW

File: pyserini/trectools/_base.py [hunk 1]
```diff
@@ -198,3 +198,61 @@ def concat(runs):
...
+        self.qrels_data = pd.DataFrame(columns=TrecQrels.columns)
+
+        if filepath is not None:
+            self.read_run(self.filepath)
+
+    def read_run(self, filepath: str):
+        self.qrels_data = pd.read_csv(filepath, sep='\s+', names=TrecQrels.columns)
+
+    def _get_relevance(self) -> Set[str]:
+        """
```
[the remark this answers is not in the document]
Comment (lintool) [reply pyserini/trectools/_base.py:225]: Also, imperative - "Return", not "Returns".

File: pyserini/trectools/_base.py [hunk 2]
```diff
@@ -198,3 +198,61 @@ def concat(runs):
...
+
+    def topics(self) -> Set[str]:
+        """
+            Returns a set with all topics.
+        """
+
+        return set(sorted(self.qrels_data["topic"].unique()))
+
+    def get_docids(self, relevance=None, topics=None) -> List[str]:
+        """Return a list of docids given the topics & relevance of interest
```
Comment (lintool) [inline pyserini/trectools/_base.py:239]: """"Return a list of docids for a topic and a relevance grade."""

Comment (lintool) [reply pyserini/trectools/_base.py:239]: Also, another common method is `get_relevant_docids`, with a particular relevance grade, default >=1.

That is, return all documents graded 1 or above. Please implement this?

File: pyserini/trectools/_base.py [hunk 3]
```diff
@@ -198,3 +198,61 @@ def concat(runs):
...
+
+        Parameters:
+        ----------
+        relevance : List[int]
+            E.g. [0, 1, 2]. If not provided, then all relevance will be returned.
+        topics : List[int]
```
Comment (lintool) [inline pyserini/trectools/_base.py:245]: This is a bit confusing... if I ask for multiple topics, I should get `Dict[topicid, Dict[docid, grade]]` right?
But if I only ask for a topic, is the type the same? Or just `Dict[docid, grade]`?

This is an API design question... but for discussion.

File: tests/test_fusion.py [hunk 4]
```diff
@@ -81,6 +81,15 @@ def test_reciprocal_rank_fusion_complex(self):
         os.remove(self.output_path)
         os.system('rm anserini.covid-r2.*')
 
+    def test_simple_qrels(self):
+        os.system('wget -q -nc https://ir.nist.gov/covidSubmit/data/qrels-rnd1.txt')
```
Comment (lintool) [inline tests/test_fusion.py:85]: we have qrels in `tools/topics-and-qrels` - no need to fetch.

File: pyserini/trectools/_base.py [hunk 5]
```diff
@@ -198,3 +198,61 @@ def concat(runs):
         run = TrecRun()
         run.run_data = run.run_data.append([run.run_data for run in runs])
         return run
+
+
+class TrecQrels:
```
[the remark this answers is not in the document]
Comment (x65han) [reply pyserini/trectools/_base.py:203]: Sounds good to me.

**Pull request merged**